### DISCUSSION TITLE
GitHub Pull Request

### DISCUSSION THREAD

**Discussion by funprogrammer89** [community/community#47087] [New to GitHub] [answered]
I did a pull request on a project and corrected a small typo in documentation. I believe I had to fork the project to be able to make the pull request. I made the commit and the pull request and then deleted the forked project from my GitHub repo. The pull request now says closed? The commit still shows, I think, and says verified. Did I do this right? If I have to fork the project to make a pull request, is it better to fork just a single file if possible?

Thanks!

**Comment by Juanescacha** [comment]
You did it good don't worry, you only got a little confusion, happened to me the first time.

The thing is, on any project you CANNOT edit directly if you try to make a change automatically github creates a fork makes the comit and sends the pull request as you saw, even if you deleted the fork or github the pull request was made correctly.

The verified means that you made the changes directly here on the website instead outside on a terminal o text editor, you did it right!

them, being that said, the repo collaborators (the one who can edit directly) and you are the only ones who can Close the pull requests (the request to made a change onto the original repo) but logically only them can acept the PR and merge that into the original branch, you can check in that PR site who did close it and more info, if you want me to check it out, share the link for that PR and I'll glad to check it out to see if I can help you noticing something besides from that.

Hopes this is clear and helps you with your quesiton, for you to understand a little better when you made a change directly it creates a temporary automatic fork, you can do it also manually edit the file propose the PR and then if you want delete or not the fork, would be the same.

If you have any doubs feel free to mention me @.

**Comment by funprogrammer89** [comment]
Awesome thanks for the GREAT feedback!

**Reply by Juanescacha** [reply]
Thanks! Glad to help, I checked also the PR and yeah, the mistake was that you deleted too son the fork before the pull was merged, so it got closed because of that automatically!

Don't forget to mark the answer to help others having the same question as yours! :wave:

**Comment by funprogrammer89** [comment]
Here is my pull request,

https://github.com/pynecone-io/pcweb/pull/10

**Comment by airtower-luna** [comment, marked as answer]
In the PR it explicitly says that it was closed because the fork was deleted. Simply don't delete the fork until your PR has been merged (or finally rejected). :slightly_smiling_face: 

The commit is still there and someone with push access to the repository could merge it manually, or you could make a new fork and pull it to there.

> I have to fork the project to make a pull request, is it better to fork just a single file if possible?

You cannot fork a single file, it's always the whole repository. You did almost everything right, just deleted the fork too early.

**Reply by queenofcorgis** [reply]
Thanks for helping out @airtower-luna and @Juanescacha for chiming in - rockstars!

**Comment by funprogrammer89** [comment]
My pull request was accepted and merged. Do I delete the forked repo now or
keep it around?

**Reply by Juanescacha** [reply]
you could do both, if you deleted you can fork it again to make a new change proposal, if you keep it, you can use it also to make more change proposals

**Comment by ranjithabb387** [comment]
I am giving the correct username with password as personal access token but still i am not able to push or pull the code please do the needful.

**Comment by funprogrammer89** [comment]
What error are you getting? Are you using git from terminal or some other
method?

**Comment by ranjithabb387** [comment]
Hi team,

I am not able to login giving the correct username and password, please do the needful.


Thanks and Regards
Ranjitha B
